common/src/main/java/com/commercetools/sunrise/payment/domain/CreatePaymentDataProvider.java

```diff
@@ -0,0 +1,35 @@
+package com.commercetools.sunrise.payment.domain;
+
+import com.commercetools.sunrise.payment.model.CreatePaymentDataProviderImpl;
+import io.sphere.sdk.customers.Customer;
+
+import javax.annotation.Nullable;
+import java.util.Optional;
+
+/**
+ * Provides a wrapper for all data that could possibly be needed by a {@link PaymentServiceProvider} implementation
+ * to create a new payment object.
+ *
+ * Created by mgatz on 7/18/16.
+ */
+public interface CreatePaymentDataProvider {
+
+    // TODO add support for more data like cart, http request, etc.
+
+    static CreatePaymentDataProvider of() {
+        return new CreatePaymentDataProviderImpl();
+    }
+
+    /**
+     * @return the attached customer
+     */
+    @Nullable
+    Optional<Customer> getCustomer();
+
+    /**
+     * Add a customer object to this data provider instance.
+     * @param c the customer object to be added
+     * @return the object itself
+     */
+    CreatePaymentDataProvider plusCustomer(Customer c);
```

Comment:
`Plus` would mean that you are adding customers, meaning that you can have many customers. If you can only have one then I would use a `setCustomer(@Nullable Customer customer)` (nullable because by the getter I understand this can be set as `null`).

But besides that, it is not very clear to me if the implementation of this class is supposed to be more like a builder (which is fine to be mutable) or more something you can store and share (which should be immutable).